components/InputStates.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * InputStates.js
+ * 
+ * Manages the input states for a component's ports
+ */
+
+var _ = require('underscore');
+
+var inputStates = [];
+
+var requests = {
+        GET_IIP_STATE: 1,
+        SET_IIP_STATE: 2,
+        GET_EDGE_STATE: 3,
+        SET_EDGE_STATE: 4 
+};
+
+/**
+ * Getter and Setter for input states
+ *
+ * If no new state is specified, this method will get the input state associated 
+ * with the inport (and in the fugure, hasing in by source node name and source port name.
+ * 
+ * If a new state is specified, it will be stored in the hash.
+ *
+ * @param inportName name of the input port whose state is being recorded
+ * @param sourceNodeName (optional) name of the node that sent the new state data
+ * @param sourcePortName (optional) name of the the source node port that sent the state data.
+ * @param newState a State object containing both a data payload and an lm.
+ *
+ * @this noflo component context
+ * @return getter path returns the current input port state state.
+ *         setter path returns the current context 
+ */ 
+module.exports = function( inportName, sourceNodeName, sourcePortName, newState ) { 
+ 
+    switch (arguments.length) { 
+
+        case requests.GET_IIP_STATE: {
+            return inputStates[inportName];
+        }
+
+        case requests.SET_IIP_STATE: {
+            inputStates[inportName] = arguments[arguments.length-1];
+            return this;
+        }
+
+        case requests.GET_EDGE_STATE: {
+            return inputStates[inportName][sourceNodeName][sourcePortName];
```

Comment:
I am not creating PreviousLMs because they are not required for the IIP sprint.  If this is a general concern about the structure, then yes, I can make it an array instead of hash of hash of hash.  I did not realize you had changed this.